src/blog/_posts/2018-06-28-opera-54.md

```diff
@@ -0,0 +1,163 @@
+hints on navigation requests. Additionally, on the first page load, this feature provides hints for all subresources of
+the page.
+
+For more on Client Hints, see [Automating Resource Selection with Client
+Hints](https://developers.google.com/web/updates/2015/09/automating-resource-selection-with-client-hints) by Ilya
+Grigorik.
+
+### Streams
+
+[Stream API](https://streams.spec.whatwg.org) support has been extended with [`TransformStream`](https://streams.spec.whatwg.org/#ts-class).
+[Transform streams](https://streams.spec.whatwg.org/#ts) enable transforming data in stream form. It can for example be
+used to pipe between a `ReadableStream` and a `WritableStream`. The following example uses a `TransformStream` to decode
+text received in a streaming response body:
+
+    function textDecodeTransform() {
+      const decoder = new TextDecoder();
+      return new TransformStream({
+          transform(chunk, controller) {
+            controller.enqueue(decoder.decode(chunk, { stream: true }));
+          }
+      });
+    }
+
+    fetch(url).then(response => {
+      // response.body is a stream of Uint8Array chunks.
+      // But if we want chunks of text:
+      const stream = response.body.pipeThrough(textDecodeTransform());
+      // …
+    });
+
+
+### Shadow DOM
+
+The [`<slot>` element](https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element) can now participate in a
+flat layout tree, with UA style `display: contents`. Before this change, a CSS selector would not match a `<slot>`
+element. Now selectors match and children will inherit from the `<slot>` element.
+
+
+## Deprecations and Interoperability Improvements
+
+### HTTP-Based Public Key Pinning is deprecated
+
+HTTP-Based Public Key Pinning (HPKP) was intended to allow websites to send an HTTP header that pins one or more of the
+public keys present in the site's certificate chain. It has very low adoption, and although it provides security against
+certificate mis-issuance, it also creates risks of denial of service and hostile pinning.
+
+To defend against certificate misissuance, web developers should use the `Expect-CT` header, including its reporting
+function. `Expect-CT` is safer than HPKP due to the flexibility it gives site operators to recover from configuration
+errors, and due to the built-in support offered by a number of CAs.
+
+We expect to remove this in Opera 56 (Chromium 69).
+
+### AppCache deprecated in Non-secure Contexts
+
+AppCache over HTTP is deprecated. AppCache is a powerful feature that allows offline and persistent access to an
+origin. Allowing AppCache to be used over non-secure contexts makes it an attack vector for cross-site scripting hacks.
+Removal is expected in Opera 56 (Chromium 69).
+
+### Layout / CSS
+
+Two Webkit-prefixed CSS properties were been removed in this release - `-webkit-box-flex-group` and
+`-webkit-box-lines`. Percent (%) values are no longer accepted by the `-webkit-line-clamp` property.
```

Comment:
Mention what should be used instead?

Reply:
Not sure there are any... (Intent thread doesn't mention any, and even no hits on Google... Presumable regular flexbox can be used instead of the *flex* one.)